kani-compiler/src/kani_middle/transform/body.rs

```diff
@@ -182,42 +206,157 @@ impl MutableBody {
         }
     }
 
+    /// Add a new call to the basic block indicated by the given index.
+    ///
+    /// The new call will have the same span as the source instruction, and the basic block
+    /// will be split. The source instruction will be adjusted to point to the first instruction in
+    /// the new basic block.
+    pub fn add_call(
+        &mut self,
+        callee: &Instance,
+        source: &mut SourceInstruction,
+        position: InsertPosition,
```

Comment:
What if we just merge these two? I.e.: Maybe make `InsertPosition` include the source instruction.

Reply:
I added `InsertPosition` as a field to `SourceOp`